macaroons.go

```diff
@@ -206,8 +206,11 @@ func (s *Server) startMacaroonService() error {
 		return fmt.Errorf("unable to unlock macaroon DB: %v", err)
 	}
 
-	// Create macaroon files for pool CLI to use if they don't exist.
-	if !lnrpc.FileExists(s.cfg.MacaroonPath) {
+	// If a macaroon string was passed in from lndclient, then we already
+	// have a macaroon for Pool and don't need to create macaroon files.
+	// Otherwise, Create macaroon files for pool CLI to use if they don't
+	// exist.
+	if macData == "" && !lnrpc.FileExists(s.cfg.MacaroonPath) {
```

Comment:
If we don't actually look at the data, couldn't this just be a boolean called "noMacaroonCreation"?

Reply:
good call